proto-derive/src/lib.rs

```diff
@@ -55,41 +55,9 @@ fn expand_domaintype(input: &syn::DeriveInput) -> TokenStream {
         }
     };
 
+    // This is what all the hubbub is about.
```

Comment:
Not entirely. I was also wondering if we need the whole proc_macro at all. 
You can replace all these two lines that look like this
```rust
#[derive(DomainType)]
#[rawtype(RawPartSetHeader)]
```
with:
`impl DomainType<MyRawType> for ActualType {}`.
As a consequence you can delete the whole macro code:

https://github.com/informalsystems/tendermint-rs/blob/e89206774e5c1939c0c03c34782735e91621fef4/proto-derive/src/lib.rs#L10-L56


And in fact the whole proto-derive crate can be deleted, without losing any functionality or any further consequences. I would argue whole code becomes easier to navigate as the macro creates some unnecessary indirection for a single line that explains what is happing.

Reply:
Yes, true. Annotations seem cleaner to me, because I don't expect to look at the derive function. Serde does the same thing and it worked out pretty well there.

Also, we need to do some kind of JSON serialization for these domain types too and I didn't have time to get into the details of how we're going to do it. Maybe the derive macro helps, maybe not. Well, currently not really, it's just syntactic sugar.

I'll open this up to the team, I'm happy to do it either way, I just really wanted to do annotations. :)